### PR TITLE
JP-2108 Asn_Lv2NRSLAMPSpectral: Allow msaspec only if msametfl is available.

### LINKED ISSUE #6084: NRS_LAMP and no MSA metafile should not make Level 2b associations
_Issue [JP-2108](https://jira.stsci.edu/browse/JP-2108) was created on JIRA by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham):_

NIRSpec NRS_LAMP exposures that do not have a MSA metafile should not make Level 2b associations.

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Closes #6084
Resolves [JP-2108](https://jira.stsci.edu/browse/JP-2108)

Checklist
- [x] Tests
- [x] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

File: jwst/associations/lib/rules_level2b.py [hunk 1]
```diff
@@ -507,9 +507,29 @@ def __init__(self, *args, **kwargs):
...
                 value='opaque'
             ),
-            DMSAttrConstraint(
-                name='opmode',
-                sources=['opmode'],
+            Constraint(
+                [
+                    Constraint(
+                        [DMSAttrConstraint(
+                            name='opmode',
```
Comment (tapastro) [inline jwst/associations/lib/rules_level2b.py:514]: So to put this logic in simple terms, this rule passes if either: the OPMODE is not MSASPEC - or - OPMODE is MSASPEC and MSAMETFL is present?

**Review by hbushouse — COMMENTED** [review]
I *think* the logic looks correct. I guess if it produces the desired result it must be correct.

Requisite reminder for a change log entry ...

File: CHANGES.rst [hunk 2]
```diff
@@ -49,6 +49,8 @@ assign_wcs
 associations
 ------------
 
+- Asn_Lv2NRSLAMPSpectral: Allow msaspec only if msametfl is available. [#6085]
```
Comment (hbushouse) [inline CHANGES.rst:52]: This needs to go up at the top under the 1.2.1 release heading

Comment (stscieisenhamer) [reply CHANGES.rst:57]: too easy to miss those headings :-(

**Review by hbushouse — APPROVED** [review]

**Pull request merged**